python/cudf/cudf/core/dataframe.py

```diff
@@ -707,9 +708,23 @@ def __init__(
         if copy is not None:
             raise NotImplementedError("copy is not currently implemented.")
         super().__init__({}, index=cudf.Index([]))
+
         if nan_as_null is no_default:
             nan_as_null = not cudf.get_option("mode.pandas_compatible")
 
+        if cudf.get_option("mode.pandas_compatible"):
+            data, data_extracted = _extract_from_proxy(data)
```

Comment:
Isn't the purpose of this PR to expose APIs to check for this kind of thing and extract ahead of time so that consuming libraries can fix them? I don't think that this call should be here. That leaks cudf.pandas details into cudf itself. If we really need to support this, it should be by overriding methods in the cudf.pandas wrapper object.

The same applies for the index and series changes below.

Reply:
Yeah I had suggested we do the extraction within cudf in https://github.com/rapidsai/cuml/issues/6232#issuecomment-2613103057, just because we don't know the scale yet of how many RAPIDS libraries could be passing cudf.pandas objects to cudf APIs, but agreed _ideally_ developers check & extract first before passing to cudf APIs

Reply:
Great thanks. I think we have to make some changes in cuml. We can't do this fully magically and I don't think that we should try.